 DO NOT MERGE: Camera: override external facing for API1 client

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index 45ef1f8..4119140 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -442,6 +442,12 @@
         mModule->getCameraInfo(cameraId, &info));
     cameraInfo->facing = info.facing;
     cameraInfo->orientation = info.orientation;
+    // CameraInfo is for android.hardware.Camera which does not
+    // support external camera facing. The closest approximation would be
+    // front camera.
+    if (cameraInfo->orientation == CAMERA_FACING_EXTERNAL) {
+        cameraInfo->orientation = CAMERA_FACING_FRONT;
+    }
     return rc;
 }
 
